executor/src/witgen/jit/compiler.rs

```diff
@@ -0,0 +1,539 @@
+            }
+        }
+        SymbolicExpression::UnaryOperation(op, inner, _) => {
+            let inner = format_expression(inner);
+            match op {
+                UnaryOperator::Neg => format!("-{inner}"),
+            }
+        }
+    }
+}
+
+fn cell_to_var_name(
+    Cell {
+        column_name,
+        row_offset,
+        ..
+    }: &Cell,
+) -> String {
+    // TODO this might still not be unique.
+    let column_name = column_name
+        .rfind("::")
+        .map_or(column_name.as_str(), |i| &column_name[i + 2..])
+        .replace("[", "_")
+        .replace("]", "_");
+    if *row_offset < 0 {
+        format!("{column_name}_u{}", -row_offset)
+    } else {
+        format!("{column_name}_d{row_offset}")
+    }
+}
+
+/// Returns the rust code containing utility functions given a first column id and a column count
+/// that is used to store the column table.
+fn util_code<T: FieldElement>(first_column_id: u64, column_count: usize) -> Result<String, String> {
+    if !(T::has_direct_repr() && (mem::size_of::<T>() == 8 || mem::size_of::<T>() == 4)) {
+        return Err(format!(
+            "Field {}not supported",
+            T::known_field()
+                .map(|f| format!("{f} "))
+                .unwrap_or_default()
+        ));
+    }
+
+    let int_type = if mem::size_of::<T>() == 8 {
+        "u64"
+    } else {
+        "u32"
+    };
+    let double_int_type = if mem::size_of::<T>() == 8 {
+        "u128"
+    } else {
+        "u64"
+    };
+    let modulus = T::modulus();
+
+    Ok(format!(
+        r#"#![allow(non_snake_case)]
+
+#[derive(Clone, Copy, Default)]
+#[repr(transparent)]
+struct FieldElement({int_type});
+
+type IntType = {int_type};
+type DoubleIntType = {double_int_type};
+const MODULUS: IntType = {modulus}_{int_type};
+
+impl std::fmt::Display for FieldElement {{
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {{
+        write!(f, "{{}}", self.0)
+    }}
+}}
+impl From<IntType> for FieldElement {{
+    #[inline]
+    fn from(i: IntType) -> Self {{
+        Self(i)
+    }}
+}}
+impl std::ops::Add for FieldElement {{
+    type Output = Self;
+    #[inline]
+    fn add(self, b: Self) -> Self {{
+        // TODO this is inefficient.
+        Self(IntType::try_from((DoubleIntType::from(self.0) + DoubleIntType::from(b.0)) % DoubleIntType::from(MODULUS)).unwrap())
+    }}
+}}
+impl std::ops::Sub for FieldElement {{
+    type Output = Self;
+    #[inline]
+    fn sub(self, b: Self) -> Self {{
+        // TODO this is inefficient.
+        Self(IntType::try_from((DoubleIntType::from(self.0) + DoubleIntType::from(MODULUS) - DoubleIntType::from(b.0)) % DoubleIntType::from(MODULUS)).unwrap())
+    }}
+}}
+impl std::ops::Mul<FieldElement> for FieldElement {{
+    type Output = Self;
+    #[inline]
+    fn mul(self, b: FieldElement) -> FieldElement {{
+        Self(IntType::try_from((DoubleIntType::from(self.0) * b.0 as DoubleIntType) % DoubleIntType::from(MODULUS)).unwrap())
```

Comment:
```suggestion
        Self(IntType::try_from((DoubleIntType::from(self.0) * DoubleIntType::from(b.0)) % DoubleIntType::from(MODULUS)).unwrap())
```
Is there a difference between the two?

Reply:
Haha interesting. I don't think so, those are primitive types.